tests/service/PatternMatchCompilationTests.fs

```diff
@@ -373,3 +373,76 @@ match A with
         "(7,19--7,20): This expression was expected to have type 'int' but here has type 'string'"
         "(6,6--6,7): Incomplete pattern matches on this expression. For example, the value 'A' may indicate a case not covered by the pattern(s)."
     ]
+    
```

Comment:
I'd like to see more testing of syntax precedence covering every possible pattern kind on left and right of `as`